### DISCUSSION TITLE
Multiple Configuration Files in CodeQL Action

### DISCUSSION THREAD

**Discussion by ArielSAdamsNASA** [community/community#27108] [Actions]
<p>I was able to use multiple configuration files for CodeQL Actions in a parent module as seen here: <a href="https://github.com/nasa/cFS/pull/254" rel="noopener nofollow ugc">https://github.com/nasa/cFS/pull/254</a>. When using the same logic and files in a submodule, CodeQL would throw an error stating “The configuration file “/home/runner/work/cFE/cFE/.github/codeql/codeql-security.yml” does not exist”. The only changes I made was renaming that same file to codeql-config.yml and it worked with no errors. The submodule PR is seen here: <a href="https://github.com/nasa/cFE/pull/1513" class="inline-onebox" rel="noopener nofollow ugc">Fix #1512, Add Coding Standard Rules in CodeQL by ArielSAdamsNASA · Pull Request #1513 · nasa/cFE · GitHub</a>.</p>
<p>My questions are:</p>
<ol>
<li>Can multiple configuration files be used in submodules?</li>
<li>Why does the submodule require the configuration file to be named codeql-config.yml?</li>
<li>Could you provide a solution to use two configuration files with unique names in the submodule, cFE?</li>
</ol>

**Comment by RasmusWL** [comment]
<p>Hi <a class="mention" href="/arielsadamsnasa">@ArielSAdamsNASA</a>, just wanted to check whether this was still a problem for you? – from the latest update on <a href="https://github.com/nasa/cFE/pull/1513#issuecomment-848999860">that PR</a>, <em>“CodeQL workflow is successful”</em>, it sounds like you have resolved the problem since then :blush:</p>
<p><em>PS. Sorry for the late reply, we must have overlooked this question :neutral_face:</em></p>

**Comment by ArielSAdamsNASA** [comment, marked as answer]
<p>Hello <a class="mention" href="/rasmuswl">@RasmusWL</a>.</p>
<p>I was able to resolve the issue by using the full path of the configuration file with the branch name as well.  For example, nasa/cFS/.github/codeql/codeql-coding-standard.yml@main. I could use any name for the configuration file as well. No worries on the late reply.</p>
